### PR TITLE
Add logic to avoid re-scanning of agents disconnected.

### PR DESCRIPTION
**Pull request by Dwordcito**
# Description

This PR Aims to solve a problem during the delayed queue processing if an agent is during a synchronization and is moved to another worker.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp [hunk 1]
```diff
@@ -73,11 +73,45 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
...
+                const auto& agent = response.front();
+                const auto& wdbAgentNodeName = agent.at("node_name").get_ref<const std::string&>();
+                const auto& wdbAgentConnectionStatus = agent.at("connection_status").get_ref<const std::string&>();
+                const auto& wdbAgentVersion = agent.at("version").get_ref<const std::string&>();
+                const auto& wdbAgentName = agent.at("name").get_ref<const std::string&>();
+                const auto& wdbAgentIp = agent.at("ip").get_ref<const std::string&>();
+
+                // If the agent is not part of a cluster or the agent is part of the cluster and the current agent
+                // node_name
+                // is the same as the server node name. Add the agent to the list of agents to be scanned.
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:91]: Quick simplification:
```suggestion
                // Add the agent to the scan list if one of the conditions is met:
                //   - It's not in a cluster 
                //   - It's in the cluster and matches the server node name.
```

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp [hunk 2]
```diff
@@ -73,11 +73,45 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
...
+                              agentId.data(),
+                              clusterStatus ? "true" : "false",
+                              clusterNodeName.data(),
+                              wdbAgentConnectionStatus.c_str(),
+                              wdbAgentNodeName.c_str());
+                }
+            }
+            catch (const std::exception& e)
+            {
+                throw std::runtime_error("Invalid agent-info response format");
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:113]: On the other catch, we return a nullptr. Shouldn't we align with that to follow the same execution path?

Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:113]: make sense

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:1]: nope, its only for delayed queue.

**Review by sebasfalcone — APPROVED** [review]
## Testing

### Re-scan agent in same node
```
2024/12/04 16:06:07 wazuh-modulesd:vulnerability-scanner[382362] buildSingleAgentListContext.hpp:93 at handleRequest(): DEBUG: Agent 002 added to the list of agents to be re-scanned
2024/12/04 16:06:08 wazuh-modulesd:vulnerability-scanner[382362] osScanner.hpp:350 at handleRequest(): DEBUG: Vulnerability scan for OS 'Ubuntu' on Agent '002' has completed.
2024/12/04 16:06:08 wazuh-modulesd:vulnerability-scanner[382362] eventDetailsBuilder.hpp:105 at handleRequest(): DEBUG: Building event details for component type: 2
```

### Not re-scan agent in different node
```consoloe
2024/12/04 16:16:07 wazuh-modulesd:vulnerability-scanner[382362] buildSingleAgentListContext.hpp:93 at handleRequest(): DEBUG: Agent 002 not added to the list of agents to be re-scanned (cluster status: true) (cluster node_name: node_1) (connection_status: inactive) (agent node_name: node_1)
```

**Pull request merged**